common/remote_object.go

```diff
@@ -189,25 +193,71 @@ func valueFromRemoteObject(ctx context.Context, robj *cdpruntime.RemoteObject) (
 	return k6ext.Runtime(ctx).ToValue(val), err
 }
 
-func handleParseRemoteObjectErr(ctx context.Context, err error, logger *logrus.Entry) {
-	var (
-		ooe *objectOverflowError
-		ope *objectPropertyParseError
-	)
-	var merr *multiError
-	if !errors.As(err, &merr) {
-		// If this panics it's a bug :)
-		k6ext.Panic(ctx, "parsing remote object value: %w", err)
-	}
-	for _, e := range merr.Errors {
-		switch {
-		case errors.As(e, &ooe):
-			logger.Warn(ooe)
-		case errors.As(e, &ope):
-			logger.WithError(ope).Error()
-		default:
-			// If this panics it's a bug :)
-			k6ext.Panic(ctx, "parsing remote object value: %w", e)
+func parseConsoleRemoteObjectPreview(logger *log.Logger, op *cdpruntime.ObjectPreview) string {
+	obj := make(map[string]string)
+	if op.Overflow {
+		logger.Warnf("parseConsoleRemoteObjectPreview", "object is too large and will be parsed partially")
+	}
+
+	for _, p := range op.Properties {
+		val := parseConsoleRemoteObjectValue(logger, p.Type, p.Subtype, p.Value, p.ValuePreview)
+		obj[p.Name] = val
+	}
+
+	bb, err := json.Marshal(obj)
+	if err != nil {
+		logger.Errorf("parseConsoleRemoteObjectPreview", "failed to marshal object to string: %v", err)
```

Comment:
Same here, return an error?

Reply:
I could return an error here. I feel that printing an error is good enough though. The new functions are solely for parsing remote objects from `console.log`, and if something fails here i think it's better to try to log something in stdout rather than give up completely.